plugins/ltac/tacentries.ml

```diff
@@ -468,7 +468,9 @@ let register_ltac local tacl =
     let () = List.iter iter_rec recvars in
     List.map map rfun
   in
-  let defs = Future.transactify defs () in
+  (* STATE XXX: Review what is going on here. Why does this needs
+     protection? Why is not the STM level protection enough? Fishy *)
+  let defs = States.with_state_protection defs () in
```

Comment:
This seems wrong: `transactify` (used to) keeps the effect on the state (if no exception is raised) while `with_state_protection` does not.

Reply:
But the main question is why do commands that live below the state layer need to mess with it? That seems very fishy, as well as the bits in funind.